tenantcostserver, filetable: replace some `*.executor`s with `InternalExecutorFactory`

ZhouXing19 opened the pull request:
This is part of migrating the existing usages of the internal executor to the new internal executor interfaces.
The changes are based on this logic: trace up the `txn` in the exec functions of the internal executor, find out where 
they are from, and ensure that an IE is created along with this txn via `planner.WithInternalExecutor` or `sqlutil.InternalExecutorFactory.TxnWithExecutor`.

The new interfaces allows the internal executor bound to the outer txn, along with txn-related metadata.

Release note: None

link Epic CRDB-19135

ZhouXing19 committed:
tenantcostserver: replace sysTableHelper.executor with InternalExecutorFactory

This is part of the migration of the existing usages of internal executor to
the new internal executor interfaces.

The new interfaces allows the internal executor bound to the outer txn, along
with txn-related metadata.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/89888)
<!-- Reviewable:end -->

ZhouXing19 committed:
sql: replace ief.RunWithouTxn() with ief.MakeInternalExecutorWithoutTxn()

For running sql statements without bound to a txn, we allow using free floating
IE. Having a `ief.MakeInternalExecutorWithoutTxn()` that returns an IE is more
natural than having the life span of the IE just within the scope of
`ief.RunWithouTxn()`.

Release note: None

ZhouXing19 committed:
filetable: replace payloadWriter.ie with InternalExecutorFactory

This is part of the migration of the existing usages of internal executor to
the new internal executor interfaces.

The new interfaces allows the internal executor bound to the outer txn, along
with txn-related metadata.

Release note: None

ZhouXing19 commented:
Friendly pint ^

ajwerner reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 4 of 4 files at r1, 11 of 11 files at r2, 1 of 1 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/89888#-:-NEaBS0i9MjqOSszDqpv:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @msbutler)



<!-- Sent from Reviewable.io -->
